test/http-api/over-ipfs-api/bootstrap.js

```diff
@@ -28,6 +28,27 @@ module.exports = (ctl) => {
         })
       })
 
+      it('prevents duplicate inserts of bootstrap peers', () => {
+        return ctl
+          .bootstrap
+          .add(validIp4)
+          .then(res => {
+            expect(res).to.be.eql({ Peers: [validIp4] })
+            return ctl.bootstrap.add(validIp4)
+          })
+          .then((res) => {
+            expect(res).to.be.eql({ Peers: [validIp4] })
+            return ctl.bootstrap.list()
+          })
+          .then((res) => {
+            expect(res).to.exist()
+            const insertPosition = res.Peers.indexOf(validIp4)
+            expect(insertPosition).to.not.equal(-1)
+            const duplicatePosition = res.Peers.indexOf(validIp4, insertPosition + 1)
+            expect(duplicatePosition).to.equal(-1)
```

Comment:
As far as I can tell, you're asserting that validIp4 is not missing from the list of peers, but never asserting that the amount of peers are correct in the test-case. Ideal, minimal test-case would start with a empty bootstrap list, add the same address twice and assert that the list only contain once element.